packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2403,23 +2313,19 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
     this.isAnimating = false;
   }
 
-  handlePartitionChange(partitions: (string | null)[]): void {
-    const { partitionColumns } = this.state;
-    if (partitionColumns.length === 0) {
-      return;
-    }
-    this.updatePartition(partitions, partitionColumns);
+  handlePartitionChange(partitions: readonly unknown[]): void {
+    const { model } = this.props;
+    model.partition = partitions;
   }
 
-  handlePartitionFetchAll(): void {
-    this.setState({
-      partitionFilters: [],
-      isSelectingPartition: false,
-    });
+  handlePartitionMerge(): void {
+    const { model } = this.props;
+    model.partition = [];
   }
 
-  handlePartitionDone(): void {
-    this.setState({ isSelectingPartition: false });
+  handlePartitionKeyTable(): void {
+    const { model } = this.props;
+    model.openPartitionKeysTable();
```

Comment:
Much like setting the partition above, this is problematic because we're not saving/restoring this state; we want the table to be in the same state as when we closed the dashboard.
I think a good model to follow here is the `rollupConfig` model, as it's doing something similar. In my general review comment I'll give some overarching guidelines.